influxql/query_manager.go

```diff
@@ -222,10 +222,12 @@ func (qm *defaultQueryManager) Query(qid uint64) (*queryTask, bool) {
 	return query, ok
 }
 
-func (qm *defaultQueryManager) waitForQuery(qid uint64, timeout time.Duration, closing <-chan struct{}, monitorCh <-chan error) {
+func (qm *defaultQueryManager) waitForQuery(qid uint64, interrupt <-chan struct{}, timeout time.Duration, closing <-chan struct{}, monitorCh <-chan error) {
 	var timer <-chan time.Time
```

Comment:
This is just me thinking out loud, but if this is called with every query it might be more efficient to have a pool of `time.Timer`s on `defaultQueryManager`, and if you have a non-zero `timeout` you would grab one and then call `timer.Reset(timeout)`.